src/SignalRServiceExtension/TriggerBindings/SignalRTriggerDispatcher.cs

```diff
@@ -0,0 +1,98 @@
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal class SignalRTriggerDispatcher : ISignalRTriggerDispatcher
+    {
+        private readonly Dictionary<(string hub, string category, string @event), SignalRMethodExecutor> _executors =
+            new Dictionary<(string, string, string), SignalRMethodExecutor>(TupleStringIgnoreCasesComparer.Instance);
+        private readonly IRequestResolver _resolver;
+
+        public SignalRTriggerDispatcher(IRequestResolver resolver = null)
+        {
+            _resolver = resolver ?? new DefaultRequestResolver();
+        }
+
+        public void Map((string hubName, string category, string @event) key, ExecutionContext executor)
+        {
+            if (!_executors.ContainsKey(key))
+            {
+                if (key.category == Constants.Category.Connections)
+                {
+                    if (key.@event == Constants.Events.Connect)
+                    {
+                        _executors.Add(key, new SignalRConnectMethodExecutor(_resolver, executor));
+                        return;
+                    }
+                    if (key.@event == Constants.Events.Disconnect)
+                    {
+                        _executors.Add(key, new SignalRDisconnectMethodExecutor(_resolver, executor));
+                        return;
+                    }
+                    throw new SignalRTriggerException($"Event {key.@event} is not supported in connections");
+                }
+                if (key.category == Constants.Category.Messages)
+                {
+                    _executors.Add(key, new SignalRInvocationMethodExecutor(_resolver, executor));
+                    return;
+                }
+                throw new SignalRTriggerException($"Category {key.category} is not supported");
+            }
+
+            throw new SignalRTriggerException(
+                $"Duplicated key parameter hub: {key.hubName}, category: {key.category}, event: {key.@event}");
+        }
+
+        public async Task<HttpResponseMessage> ExecuteAsync(HttpRequestMessage req, CancellationToken token = default)
+        {
+            // TODO: More details about response
+            if (!_resolver.ValidateContentType(req))
+            {
+                return new HttpResponseMessage(HttpStatusCode.UnsupportedMediaType);
+            }
+
+            if (!TryGetDispatchingKey(req, out var key))
+            {
+                return new HttpResponseMessage(HttpStatusCode.BadRequest);
+            }
+            
+            if (_executors.TryGetValue(key, out var executor))
+            {
+                try
+                {
+                    return await executor.ExecuteAsync(req);
+                }
+                catch (SignalRTriggerException ex)
```

Comment:
>SignalRTriggerException [](start = 23, length = 23)

Could some general exception throws from `ITriggeredFunctionExecutor`? Shall we handle as well?

Reply:
Updated. We will handle different details exceptions and response different result